crates/sui-core/src/authority/authority_store_tables.rs

```diff
@@ -0,0 +1,335 @@
+    /// epoch change.
+    pub(crate) consensus_message_processed: DBMap<TransactionDigest, bool>,
+
+    // Tables used for authority batch structure
+    /// A sequence on all executed certificates and effects.
+    pub executed_sequence: DBMap<TxSequenceNumber, ExecutionDigests>,
+
+    /// A sequence of batches indexing into the sequence of executed transactions.
+    pub batches: DBMap<TxSequenceNumber, SignedBatch>,
+
+    /// The following table is used to store a single value (the corresponding key is a constant). The value
+    /// represents the index of the latest consensus message this authority processed. This field is written
+    /// by a single process acting as consensus (light) client. It is used to ensure the authority processes
+    /// every message output by consensus (and in the right order).
+    pub(crate) last_consensus_index: DBMap<u64, ExecutionIndices>,
+
+    /// Map from each epoch ID to the epoch information. The epoch is either signed by this node,
+    /// or is certified (signed by a quorum).
+    pub(crate) epochs: DBMap<EpochId, AuthenticatedEpoch>,
+}
+impl<S: Eq + Debug + Serialize + for<'de> Deserialize<'de>> StoreTables<S> {
+    /// If with_secondary_path is set, the DB is opened in read only mode witht he path specified
+    pub fn open_impl<P: AsRef<Path>>(
+        path: P,
+        db_options: Option<Options>,
+        with_secondary_path: Option<P>,
+    ) -> Self {
+        let (options, point_lookup) = default_db_options(db_options, None);
+
+        let db = {
+            let path = &path;
+            let db_options = Some(options.clone());
+            let opt_cfs: &[(&str, &rocksdb::Options)] = &[
+                (OBJECTS_TABLE_NAME, &point_lookup),
+                (TX_TABLE_NAME, &point_lookup),
+                (OWNER_INDEX_TABLE_NAME, &options),
+                (CERTS_TABLE_NAME, &point_lookup),
+                (PENDING_EXECUTION, &options),
+                (PARENT_SYNC_TABLE_NAME, &options),
+                (EFFECTS_TABLE_NAME, &point_lookup),
+                (ASSIGNED_OBJECT_VERSIONS_TABLE_NAME, &options),
+                (NEXT_OBJECT_VERSIONS_TABLE_NAME, &options),
+                (CONSENSUS_MESSAGE_PROCESSED_TABLE_NAME, &options),
+                (EXEC_SEQ_TABLE_NAME, &options),
+                (BATCHES_TABLE_NAME, &options),
+                (LAST_CONSENSUS_TABLE_NAME, &options),
+                (EPOCH_TABLE_NAME, &point_lookup),
+            ];
+            if let Some(p) = with_secondary_path {
+                typed_store::rocks::open_cf_opts_secondary(path, Some(&p), db_options, opt_cfs)
+            } else {
+                typed_store::rocks::open_cf_opts(path, db_options, opt_cfs)
+            }
+        }
+        .expect("Cannot open DB.");
+
+        let executed_sequence =
+            DBMap::reopen(&db, Some(EXEC_SEQ_TABLE_NAME)).expect("Cannot open CF.");
+
+        let (
```

Comment:
This opening and naming pattern is used in many places around the code. Investigating macros we can provide from mysten-infra to reduce the noise